pyrsp/utils.py

```diff
@@ -31,6 +31,21 @@ def unpack(pkt):
     pkt = pkt[1:-3]
     return pkt
 
+def decode_data(data):
+    """ decodes data from received packet, decoding algorithm is described
+here: https://sourceware.org/gdb/onlinedocs/gdb/Overview.html
+    """
+    decoded_data = ""
+    j = -1
+    for i, ch in enumerate(data):
+        if ch == '*' and data[i - 1] != '*':
+            n = ord(data[i + 1]) - 29
+            decoded_data = decoded_data + (data[i - 1] * n)
+            j = i + 1
+        elif i != j:
+            decoded_data = decoded_data[::] + ch
```

Comment:
isn't the [::] superflous here <str> + <str> creates a completely new string anyway. no?